daemons/gptp/common/ieee1588port.cpp

```diff
@@ -1196,23 +1253,42 @@ void IEEE1588Port::processEvent(Event e)
 				sendSignalMessage = true;
 			}
 
-			if (sendSignalMessage) {
-				if (!isGM) {
-				// Send operational signalling message
-					PTPMessageSignalling *sigMsg = new PTPMessageSignalling(this);
+			if (sendSignalMessage && clock->getStaticPortState() != PTP_MASTER) {
```

Comment:
I'm not sure if this conditional is correct. Should this run regardless of whether static port configuration is set or not? If so, you might want to key off the current port state rather than the static port state.

Reply:
I came to the same conclusion. Changed it to use the current port state.